x-pack/plugins/observability_ai_assistant/public/analytics/schema.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { RootSchema } from '@kbn/analytics-client';
+import { Message } from '../../common';
+import type { Feedback } from '../components/feedback_buttons';
+
+export const MESSAGE_FEEDBACK = 'message_feedback';
+
+export interface MessageFeedback extends Message {
+  feedback: Feedback;
+}
+
+export interface TelemetryEvent {
+  eventType: typeof MESSAGE_FEEDBACK;
+  schema: RootSchema<MessageFeedback>;
+}
+
+export const MESSAGE_FEEDBACK_SCHEMA: TelemetryEvent = {
+  eventType: MESSAGE_FEEDBACK,
+  schema: {
+    '@timestamp': {
+      type: 'text',
+      _meta: {
+        description: 'The timestamp of the message.',
+      },
+    },
+    feedback: {
+      type: 'text',
+      _meta: {
+        description: 'Whether the user has deemed this response useful or not',
+      },
+    },
+    message: {
+      properties: {
+        content: {
+          type: 'text',
+          _meta: {
+            description: 'The response generated by the LLM.',
+            optional: true,
+          },
+        },
+        name: {
+          type: 'text',
+          _meta: {
+            description: 'The name of the function that was executed.',
```

Comment:
This seems to be the same as for `function_call`?

Reply:
Yes. That's primarily due to the awkwardness in the data model that OpenAI specified for its function handling, specifically `name` can sometimes point to the name of the function being called.